lib/pr_checker.js

```diff
@@ -169,9 +169,30 @@ class PRChecker {
       logger.warn('No full CI runs detected');
     }
 
+    let lastCiDate = null;
     for (const [type, ci] of ciMap) {
       const name = CI_TYPES.get(type).name;
       logger.info(`Last ${name} CI on ${ci.date}: ${ci.link}`);
+      if (!lastCiDate || lastCiDate < ci.date) {
+        lastCiDate = ci.date;
+      }
+    }
+
+    const commitsAfterCi = [];
+    if (lastCiDate) {
+      commits.forEach((c) => {
+        const commit = c.commit;
+        if (commit.committedDate > lastCiDate) {
+          commitsAfterCi.push(commit);
```

Comment:
I believe this could be refactored to be more like:

```js
if (lastCiDate) {
  let didLogHeader = false;
  commits.forEach((c) => {
    const commit = c.commit;
    if (commit.committedDate > lastCiDate) {
      if (!didLogHeader) {
        logger.warn('Commits pushed after the last CI run:');
        status = false;
        didLogHeader = true;
      }
      logger.warn(commit.message);
    }
  });
}
```

(Note also the addition of `status = false` since these new warnings technically mean that the CI check isn't clean.)

Reply:
right , I will fix it